### PR TITLE
Add types for react-audio-player

### PR DESCRIPTION
**Pull request by dpgil**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by peterblazejewicz — CHANGES_REQUESTED** [review]
@dpgil it looks really good!
What about exposing public ref to audio element and to public methods:
https://github.com/justinmc/react-audio-player/blob/master/src/index.jsx#L66-L86
https://github.com/justinmc/react-audio-player#access-to-the-audio-element
Except of that advanced props, looks good!

**Review by peterblazejewicz — CHANGES_REQUESTED** [review]
~~removed duplicate~~

**Comment by dpgil** [comment]
Thanks @peterblazejewicz ! I went ahead and allowed access to the audio element as a variable on the class

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@dpgil thx for the contribution!

File: types/react-audio-player/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    onVolumeChanged?: (event: React.SyntheticEvent<HTMLAudioElement, Event>) => void;
+    onLoadedMetadata?: (event: React.SyntheticEvent<HTMLAudioElement, Event>) => void;
+    preload?: Preload;
+    src?: string;
+    style?: React.CSSProperties;
+    title?: string;
+    volume?: number;
+}
+
+export class ReactAudioPlayer extends React.Component<ReactAudioPlayerProps, any> {
```
Comment (uniqueiniquity) [inline types/react-audio-player/index.d.ts]: This should use `export default`.

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

**Comment by peterblazejewicz** [comment]
> @dpgil - It appears Travis did not correctly run on this PR! /cc ..  to investigate and advise.

I think it was a CI problem, iv'e run that PR locally and it passes lint/tests.
So just ignore that, just
/cc @uniqueiniquity

**Review by uniqueiniquity — APPROVED** [review]

**Comment by uniqueiniquity** [comment]
Congratulations on your first DefinitelyTyped contribution!
Thank you for being a part of the community!

**Pull request merged**